packages/trader/src/Stores/Modules/Trading/Helpers/chart.ts

```diff
@@ -0,0 +1,166 @@
+            payload.data.action = open_close_action;
+            return payload;
+        } else if (state === STATE_TYPES.CHART_TYPE_CHANGE) {
+            payload.data.action = 'choose_chart_type';
+            return payload;
+        } else if (state === STATE_TYPES.CHART_INTERVAL_CHANGE) {
+            payload.data.action = 'choose_time_interval';
+            return payload;
+        }
+    }
+    if (indicator_type_name && indicators_category_name) {
+        const payload = {
+            data: {
+                action: '',
+                indicator_type_name,
+                indicators_category_name,
+            },
+            event_type: indicators_event_type,
+        };
+        if (state === STATE_TYPES.INDICATOR_DELETED) {
+            payload.data.action = 'delete_active';
+            return payload;
+        } else if (state === STATE_TYPES.INDICATOR_SETTINGS_OPEN) {
+            payload.data.action = 'edit_active';
+            return payload;
+        } else if (state === STATE_TYPES.INDICATOR_INFO_TOGGLE) {
+            payload.data.action = is_info_open ? 'info_open' : 'info_close';
+            return payload;
+        }
+    }
+    if (state === STATE_TYPES.INDICATORS_MODAL_TOGGLE && 'is_open' in option) {
+        return {
+            data: { action: open_close_action },
+            event_type: indicators_event_type,
+        };
+    }
+    if (state === STATE_TYPES.INDICATOR_ADDED && indicator_type_name) {
+        return {
+            data: {
+                action: 'add_active',
+                indicator_type_name,
+                indicators_category_name,
+                subform_name: is_info_open ? 'indicators_info' : 'indicators_type',
+            },
+            event_type: indicators_event_type,
+        };
+    }
+    if (state === STATE_TYPES.INDICATORS_CLEAR_ALL) {
+        return {
+            data: { action: 'clean_all_active' },
+            event_type: indicators_event_type,
+        };
+    }
+    if (option.search_string && (state === STATE_TYPES.MARKET_SEARCH || state === STATE_TYPES.INDICATOR_SEARCH)) {
+        return {
+            data: {
+                action: 'search',
+                search_string,
+            },
+            event_type: state === STATE_TYPES.MARKET_SEARCH ? market_event_type : indicators_event_type,
+        };
+    }
+    if (state === STATE_TYPES.MARKETS_LIST_TOGGLE && 'is_open' in option) {
+        return {
+            data: {
+                action: open_close_action,
+                market_type_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.SYMBOL_CHANGE && market_type_name) {
+        return {
+            data: {
+                action: 'choose_market_type',
+                market_type_name,
+                tab_market_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.MARKET_INFO_REDIRECT && tab_market_name) {
```

Comment:
INSTRUMENT_INFO_REDIRECT if we are going to follow change i suggested.

Reply:
@henry-deriv I'm calling is market in order to have the same definition of symbol as marketing team is calling it in my task. Please see https://github.com/binary-com/deriv-app/pull/9848#discussion_r1316889723